### PR TITLE
feat(@deepnote/convert): export detectFormat and NotebookFormat

### PR DESCRIPTION
**Pull request by saltenasl**
## Summary by CodeRabbit

* **New Features**
  * Format detection now recognizes Jupyter, Deepnote, Marimo, Percent, and Quarto formats and is exposed via the package API.
* **Tests**
  * Added comprehensive tests covering format detection, edge cases, and error handling.
* **Chores**
  * Package version bumped to 2.1.0.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/convert/src/format-detection.test.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    it('throws error for unsupported .py file format', () => {
+      const content = `print("hello")
+x = 1
+`
+      expect(() => detectFormat('notebook.py', content)).toThrow(
+        'Unsupported Python file format. File must be percent format (# %%) or Marimo (@app.cell).'
+      )
+    })
+  })
```
[the remark this answers is not in the document]
Comment (saltenasl) [reply packages/convert/src/format-detection.test.ts:147]: what's the probability this will happen in a real-world scenario?
